File: algorithms/001_two_sum.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9))
}

func twoSum(nums []int, target int) []int {
	for i, l := 0, len(nums); i < l; i++ {
		for j := i + 1; j < l; j++ {
			if t := nums[i] + nums[j]; t == target {
				return []int{i, j}
			}
		}
	}

	return []int{}
}

File: algorithms/001_two_sum.php
<?php
declare(strict_types=1);

/**
 * @param int[] $nums
 * @param int $target
 * @return int[]
 */
function twoSum(array $nums, int $target): array
{
    foreach ($nums as $key1 => $num1) {
        foreach ($nums as $key2 => $num2) {
            if (($num1 + $num2) === $target) {
                return [$key1, $key2];
            }
        }
    }

    return [];
}

var_dump(twoSum([2, 7, 11, 15], 9));
